File: packages/telemetry/typescript/examples/test_cohere.ts
/**
 * Cohere — Latitude telemetry example.
 *
 * Required env vars:
 * - LATITUDE_API_KEY
 * - LATITUDE_PROJECT_SLUG
 * - COHERE_API_KEY
 *
 * Install: npm install cohere-ai @traceloop/instrumentation-cohere
 */

import { randomUUID } from "node:crypto"
import * as CohereSDK from "cohere-ai"
import { CohereClient } from "cohere-ai"
import { capture, Latitude } from "../src"
import { createCohereInstrumentation } from "../src/instrumentations/cohere.ts"

const latitude = new Latitude({
  apiKey: process.env.LATITUDE_API_KEY!,
  project: process.env.LATITUDE_PROJECT_SLUG!,
  disableBatch: true,
  instrumentations: [createCohereInstrumentation(CohereSDK)],
})

const PROVIDER = "cohere"
const MODEL = "command-r"
const SESSION_ID = `${PROVIDER}-${randomUUID().slice(0, 8)}`

function ctx(scenario: string, ...extraTags: string[]) {
  return {
    tags: ["example", PROVIDER, ...extraTags],
    sessionId: SESSION_ID,
    userId: "example-user",
    metadata: { scenario, environment: "local" },
  }
}

async function chat() {
  const client = new CohereClient({
    token: process.env.COHERE_API_KEY,
  })

  const response = await client.chat({
    model: MODEL,
    message: "Say 'Hello from Cohere!' in exactly 5 words.",
    maxTokens: 50,
  })

  return response.text
}

async function main() {
  await latitude.ready

  await chat()

  await capture("cohere-chat-capture", chat, ctx("chat"))

  await latitude.flush()
  await latitude.shutdown()
}

main().catch(console.error)

File: packages/telemetry/python/examples/test_cohere.py
"""
Cohere — Latitude telemetry example.

Required env vars:
- LATITUDE_API_KEY
- LATITUDE_PROJECT_SLUG
- COHERE_API_KEY

Install: uv add cohere
"""

import os
import uuid

import cohere

from latitude_telemetry import Latitude, capture

latitude = Latitude(
    api_key=os.environ["LATITUDE_API_KEY"],
    project=os.environ["LATITUDE_PROJECT_SLUG"],
    instrumentations={"cohere": cohere},
    disable_batch=True,
)

PROVIDER = "cohere"
MODEL = "command-a-03-2025"
SESSION_ID = f"{PROVIDER}-{uuid.uuid4().hex[:8]}"


def _ctx(scenario: str, *extra_tags: str) -> dict:
    return {
        "tags": ["example", PROVIDER, *extra_tags],
        "session_id": SESSION_ID,
        "user_id": "example-user",
        "metadata": {"scenario": scenario, "environment": "local"},
    }


def chat() -> str:
    client = cohere.Client(api_key=os.environ["COHERE_API_KEY"])

    response = client.chat(
        model=MODEL,
        message="Say 'Hello from Cohere!' in exactly 5 words.",
        max_tokens=50,
    )

    return response.text


if __name__ == "__main__":
    chat()

    capture("cohere-chat-capture", chat, _ctx("chat"))

    latitude.flush()
